Application/src/Utils/Structure.js

```diff
@@ -67,7 +65,34 @@ export async function getImageGrid(image) {
   let squareWidth = width / 3;
   let squareHeight = height / 3;
 
-  newImages.topLeft.img = ogImage
+  newImages = await bigGrid(newImages, ogImage, squareHeight, squareWidth);
+  newImages = await smallGrid(newImages, ogImage, squareHeight, width);
+
```

Comment:
The logic appears to be just right for a two object approach to this problem. Also why are you assigning newImages to the value of the bigGrid function when you are not doing anything with it besides reassigning it to the value of the smallGrid function. Why not just assign newImages to smallGrid in the first place.

Reply:
I am using the 'bigGrid' (3x3) for the color rating. The 'smallGrid' (3x1) is used for the text rating.

Reply:
And the attributes for newImages are not getting overwritten? From one function to the next. Can you please check that